vivisect/base.py

```diff
@@ -212,9 +216,23 @@ def _handleADDSEGMENT(self, einfo):
         self.segments.append(einfo)
 
     def _handleADDRELOC(self, einfo):
-        self.reloc_by_va[einfo[0]] = einfo[1]
+        rva, rtype = einfo
+        self.reloc_by_va[rva] = rtype
         self.relocations.append(einfo)
 
+        if rtype == RTYPE_BASERELOC:
+            fnm = self.getFileByVa(rva)
+            imgbase = self.getFileMeta(fnm, 'imagebase')
+            rbase = imgbase
+
+            ptr = self.readMemoryPtr(rva)
+            ptr += rbase
+            ptr &= e_bits.u_maxes[self.psize]
+            self.writeMemoryPtr(rva, ptr)
+
+        logger.info('_handleADDRELOC: %x -> %x (map: 0x%x)', rva, ptr, rbase)
```

Comment:
Should this logging message be underneath the if? Becauseptr and rbase are only defined when that if block is hit.

Reply:
Also, how potentially noisy is this?

Reply:
it can be noisy if you have debugging turned up.  otherwise, it does nothing.
it's noisy for my test case, which as ~10k relocs (27MB bin).  but otherwise, not too bad.